aws/table_aws_spot_price_history.go

```diff
@@ -0,0 +1,151 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+	"time"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsSpotPriceHistory(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_spot_price_history",
+		Description: "AWS Spot Price History",
+		List: &plugin.ListConfig{
+			Hydrate: listSpotPriceHistory,
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "availability_zone", Require: plugin.Optional},
+				{Name: "instance_type", Require: plugin.Optional},
+				{Name: "product_description", Require: plugin.Optional},
+				{Name: "start_time", Require: plugin.Optional},
+				{Name: "end_time", Require: plugin.Optional},
+			},
+		},
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{Name: "availability_zone", Description: "The Availability Zone.", Type: proto.ColumnType_STRING},
+			{Name: "instance_type", Description: "The instance type.", Type: proto.ColumnType_STRING},
+			{Name: "product_description", Description: "A general description of the AMI.", Type: proto.ColumnType_STRING},
+			{Name: "spot_price", Description: "The maximum price per unit hour that you are willing to pay for a Spot Instance.", Type: proto.ColumnType_STRING},
+			{Name: "timestamp", Description: "The date and time the request was created", Type: proto.ColumnType_TIMESTAMP},
+			{Name: "start_time", Description: "The date and time, up to the past 90 days, from which to start retrieving the price history data.", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromQual("start_time")},
+			{Name: "end_time", Description: "The date and time, up to the current date, from which to stop retrieving the price history data.", Type: proto.ColumnType_TIMESTAMP, Transform: transform.FromQual("end_time")},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listSpotPriceHistory(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Create Session
+	svc, err := EC2Client(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_spot_price_history.listSpotPriceHistory", "connection_error", err)
+		return nil, err
+	}
+	if svc == nil {
+		// Unsupported region, return no data
+		return nil, nil
+	}
+
+	// Limiting the results
+	maxItems := int32(100)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxItems {
+			if limit < 20 {
+				maxItems = 20
+			} else {
+				maxItems = limit
+			}
+		}
+	}
+
+	input := ec2.DescribeSpotPriceHistoryInput{
+		MaxResults: &maxItems,
+	}
+
+	// use filters instead of availability_zone which doesn't work
```

Comment:
For building filters, can you please see some other examples like https://github.com/turbot/steampipe-plugin-aws/blob/d4b7413a25eed69b1a9057f8d568df272fb16512/aws/table_aws_rds_db_instance.go#L468 and https://github.com/turbot/steampipe-plugin-aws/blob/d4b7413a25eed69b1a9057f8d568df272fb16512/aws/table_aws_ec2_instance.go#L450?

Reply:
Finally, i removed filters. I don't understand what the goal of theese filters on the API.
All fields seem to be present, as a query parameter, so i prefer to delete it.

If you understand what's the goal about these filters, feel free to ask me to put it here

Reply:
@aminvielledebatAtBedrock This filter is a bit confusing, as it seems to have a lot of crossover with the other top level input params the API supports, like for the availability zone, product description, and timestamp.

For the common filters, e.g., availability zone, product description, instance type, do you know if the `filter` input params provides additional or different filtering capabilities than the top level input params?

If not, then if the only additional filter it offers is for spot price, I think I'd be ok to not use it for now, and use the other top level input params like you already are.

Reply:
Yes I think too @cbruno10 in our context.

> For the common filters, e.g., availability zone, product description, instance type, do you know if the filter input params 
   provides additional or different filtering capabilities than the top level input params?

For me not. The only interesting thing by using filters, is the ability to pass many values in one filter (for instance, many instance types). But it's only true in Go, not with SQL (or i don't know how to do that :panda_face: ).
For example, a query with this kind of where clause `where instance_type like any (array['m5.4xlarge', 'm6g.4xlarge'])` doesn't use this ability.

The `qual["instance_type"]` is not filled in the table method :cry: .

IMO filters in this table are not relevant

Reply:
@aminvielledebatAtBedrock When using `like`, the key columns are passed in. This is intentional, so the alternative would be to use the SQL `in` operator, which would work I believe.

Reply:
That being said, I agree, `filters` isn't useful for now, so we should avoid adding support for `filters` for now.

Reply:
Cool ! Thank you for your time ;-)